google-home-community.groovy

```diff
@@ -411,6 +430,142 @@ def deviceTraitDelete(deviceTrait) {
+        "armed away":            "armed away",
+    ]
+
+    section("Arm/Disarm Settings") {
+        input(
+            name: "${deviceTrait.name}.armedAttribute",
+            title: "Armed/Disarmed Attribute",
+            type: "text",
+            defaultValue: "securityKeypad",
+            required: true
+        )
+        input(
+            name: "${deviceTrait.name}.armLevelAttribute",
+            title: "Current Arm Level Attribute",
+            type: "text",
+            defaultValue: "securityKeypad",
+            required: true
+        )
+        input(
+            name: "${deviceTrait.name}.exitAllowanceAttribute",
+            title: "Exit Delay Value Attribute",
+            type: "text",
+            defaultValue: "exitAllowance",
+            required: true
+        )
+        input(
+            name: "${deviceTrait.name}.cancelCommand",
+            title: "Cancel Arming Command",
+            type: "text",
+            defaultValue: "disarm",
+            required: true
+        )
+        input(
+            name: "${deviceTrait.name}.armLevels",
+            title: "Supported Alarm Levels",
+            type: "enum",
+            options: hubitatAlarmLevels,
+            multiple: true,
+            required: true,
+            submitOnChange: true
+        )
+
+        deviceTrait.armLevels.each { armLevel ->
+            input(
+                name: "${deviceTrait.name}.armLevels.${armLevel.key}.googleNames",
+                title: "Google Home Level Names for ${hubitatAlarmLevels[armLevel.key]}",
+                description: "A comma-separated list of names that the Google Assistant will " +
+                             "accept for this alarm setting",
+                type: "text",
+                required: "true",
+                defaultValue: hubitatAlarmLevels[armLevel.key]
+            )
+
+            input(
+                name: "${deviceTrait.name}.armCommands.${armLevel.key}.commandName",
+                title: "Hubitat Command for ${hubitatAlarmLevels[armLevel.key]}",
+                type: "text",
+                required: "true",
+                defaultValue: hubitatAlarmCommands[armLevel.key]
+            )
+
+            input(
+                name: "${deviceTrait.name}.armValues.${armLevel.key}.value",
+                title: "Hubitat Value for ${hubitatAlarmLevels[armLevel.key]}",
+                type: "text",
+                required: "true",
+                defaultValue: hubitatAlarmValues[armLevel.key]
+            )
+        }
+    }
+    section("Arm/Disarm Security") {
+        input(
+            name: "${deviceTrait.name}.useDevicePinCodes",
+            title: "Select to use device pincodes.  Deselect to use Google Home Community app pincodes.",
+            type: "bool",
+            defaultValue: false,
+            required: true,
+            submitOnChange: true
+        )
+        if (deviceTrait.useDevicePinCodes == true) {
```

Comment:
Could this be configured at the device type level?  It seems like this should be a selectable option for any trait.

Reply:
Good point.  looking at this.

Reply:
Right now this is at the trait level.  You would prefer it up one level at the device level?  When the user is adding traits to a device?

Reply:
Yeah, it doesn't seem like it should trait-specific to me.  Any device could offer device-specific PIN codes in theory, and I think it would make more sense as a peer of the existing PIN code settings at the device type level.

Reply:
I agree.  The initial user story I was thinking of was if a device had a lock + armDisarm trait, then they could have 2 sets of device level pins.  Thinking about it, that's probably a highly unlikely scenario.

I'll move it up a level with a toggle to use them from the device driver context, or the device context in GHC.

Reply:
Moved this to the device level from the trait level.  There is a boolean slider to allow the matched index to be send back as a command parameter:
                        input(
                            name: "${devicePropertyName}.returnUserIndexToDevice",
                            title: "Select to return the user index with the device command on a pincode match. " +
                                    "Not all device drivers support this operation.",
                            type: "bool",
                            defaultValue: false,
                            submitOnChange: true
                        )

This is available on all devices, but I only implemented it on the arm trait and lock trait.  What are your thoughts on this?  I can add it to all if think it's required,  Just becomes a bit ore complicated on devices that already pass a parameter back (such as brightness).

Reply:
Moved this setting to the arm and lock traits only.